.github/workflows/test.yml

```diff
@@ -0,0 +1,46 @@
+name: Build
+on: [pull_request]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        image:
+          - juno-stable
+          - juno-stable-installer
+          - juno-unstable
+          - juno-unstable-installer
+          - loki-stable
+          - loki-unstable
+          - stable
+          - stable-installer
+          - unstable
+          - unstable-installer
+          - latest
+
+    name: ${{ matrix.image }}
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Clone
+      uses: actions/checkout@v1
+
+    - name: Build Docker Image
+      run: |
+        docker login -u "${{ secrets.DockerUsername }}" -p "${{ secrets.DockerPassword }}"
+        IMAGE="${{ matrix.image }}"
+        IMAGE_NAME="$(echo "$IMAGE" | awk -F'-' '{print $1}')"
+        # cut the first hyphenated section from tag name, since it will be the image name instead
+        TAG=${IMAGE#*-}
+        cd "$IMAGE" || exit 1
+        # don't build default tags.
+        declare -a arr=("stable" "stable-installer" "unstable" "unstable-installer" "latest")
+        if [[ ! "${arr[@]}" =~ "$IMAGE" ]]; then
+          # rename unstable tags to latest
+          if [ "$TAG" == "unstable" ]; then TAG="latest"; fi
+          if [ "$TAG" == "unstable-installer" ]; then TAG="installer"; fi
+          docker build -t elementary/"$IMAGE_NAME":"$TAG" .
+        fi
```

Comment:
Considering we are not actually publishing anything here, we could simply just do `docker build -t "${{ matrix.image }}" -f "${{ matrix.image }} ." and not do all of the rename stuff.

Reply:
i was lazy and copy-pasted. :scream: I've updated the job to just run the builds.